components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -706,32 +706,33 @@ private String getCaptchaParams(String tenantDomain, int failedLoginAttempts) {
                             }
                         }
                     } else if (captchaEnabledConfigName.equals(connectorConfig.getName())) {
-                        if (Boolean.parseBoolean(connectorConfig.getValue())) {
-                            Property[] maxFailedConfig = BasicAuthenticatorDataHolder.getInstance()
-                                    .getIdentityGovernanceService()
-                                    .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName}, tenantDomain);
-                            Property maxFailedProperty = maxFailedConfig[0];
-                            try {
-                                if (Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts) {
-                                    if (log.isDebugEnabled()) {
-                                        log.debug("Number of failed attempts is higher than max failed " +
-                                                "attempts for reCaptcha. Recaptcha will be enforced");
-                                    }
-                                    captchaParams += BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
-                                } else {
-                                    if (log.isDebugEnabled()) {
-                                        log.debug("Number of failed attempts is less than max failed " +
-                                                "attempts for reCaptcha. Recaptcha will not be enforced");
-                                    }
+                        if (!Boolean.parseBoolean(connectorConfig.getValue())) {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Enforcing recaptcha for SSO Login is not enabled.");
+                            }
+                            continue;
+                        }
+                        Property[] maxFailedConfig =
+                                BasicAuthenticatorDataHolder.getInstance().getIdentityGovernanceService()
+                                        .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName},
+                                                tenantDomain);
+                        Property maxFailedProperty = maxFailedConfig[0];
+                        try {
+                            if (Integer.valueOf(maxFailedProperty.getValue()) >= failedLoginAttempts) {
+                                if (log.isDebugEnabled()) {
+                                    log.debug("Number of failed attempts is less than max failed " +
+                                            "attempts for reCaptcha. Recaptcha will not be enforced");
```

Comment:
Condition says `Integer.valueOf(maxFailedProperty.getValue()) >= failedLoginAttempts`

Reply:
Condition is correct right? It is the negation of the previous condition `Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts`.  We check if the max failed attempts is higher or equals to failed login attempts and indicate recaptcha will not be enforced.